code/game/objects/items/weapons/policetape.dm

```diff
@@ -162,10 +162,13 @@ var/list/tape_roll_applications = list()
 					for(var/obj/structure/window/W in T)
 						if(W.is_fulltile() || W.dir == reverse_dir[dir])
 							possible_dirs += dir
+			for(var/obj/structure/window/window in start)
+				if(istype(window) && !window.is_fulltile())
+					possible_dirs += window.dir
```

Comment:
Possible_dirs is using dirflsgs, so shouldn't it use || instead of += to prevent double-counting a dir?

Reply:
And shouldn't it also specifically exclude the dir of the window on this turf? Like, if there's a window facing west, you cant take tape in that direction

Reply:
This area of code specifically is responsible for placing tape on a single tile, so starting and ending without moving. As a result it cant really be blocked by that window, and code is mostly there to calculate way it should appear.

As for being binary, some testing showed you are right. Was mostly going off the way it was done in nearby part of the file. Fixed that both in new and old.